docs/application/web/guides/machine-learning/trainer.md

```diff
@@ -0,0 +1,294 @@
+# Trainer
+
+ML Trainer API allows you to construct, control, and train a machine learning model on Tizen devices.
+
+The main features of the ML Trainer API include the following:
+
+- Constructing deep neural network (DNN)
+   - You can construct a model directly in the code or with a description file.
+
+- Training with your own data
+   - ML Trainer API allows you to train a model using data loaded from a file.
+
+- Evaluating the model during training
+   - You can easily validate and test your model during the training process by defining the dataset.
+
+> [!NOTE]
+> For clarity, errors in the examples are not handled correctly. In production code, take care of all of them to prevent your application from terminating. All errors can be found in [API reference](../../api/latest/device_api/mobile/tizen/ml_trainer.html).
+
+## Prerequisites
+
+To use ML Trainer API, include the following features in your `config.xml` file:
+
+```xml
+<feature name="http://tizen.org/feature/machine_learning"/>
+<feature name="http://tizen.org/feature/machine_learning.training"/>
+```
+
+In case of saving or loading model files from the outside of the application's own resources, the application has to request permission by adding the following privileges to the `config.xml` file:
+
+```xml
+<!-- For accessing media storage -->
+<tizen:privilege name="http://tizen.org/privilege/mediastorage"/>
+<!-- For accessing external storage -->
+<tizen:privilege name="http://tizen.org/privilege/externalstorage"/>
+<!-- For reading and writing -->
+<tizen:privilege name="http://tizen.org/privilege/filesystem.read"/>
+<tizen:privilege name="http://tizen.org/privilege/filesystem.write"/>
+```
+
+## Build blocks
+
+There are four major components of ML Trainer API, all of which are listed below:
+- Model
+- Layer
+- Optimizer
+- Dataset
+
+Let's explore all of them with a simple example.
+
+## Create dataset
+
+Dataset is in charge of feeding data into the model. ML Trainer API supports loading datasets from a file. The file format is described in [the guide for Tizen Native API](../../../native/guides/machine-learning/machine-learning-training#set-a-dataset-from-file):
+
+```javascript
+var trainFile = "documents/train.dat";
+var validateFile = null;
+var testFile = "documents/test.dat";
+var dataset = tizen.ml.trainer.createFileDataset(trainFile, validateFile, testFile);
+```
+
+Optionally you can [set properties](../../api/7.0/device_api/mobile/tizen/ml_trainer.html#Dataset::setProperty) of the dataset with `setProperty` method:
```

Comment:
Change has been suggested to maintain consistency with other API links on this page.
```suggestion
Optionally you can [set properties](../../api/latest/device_api/mobile/tizen/ml_trainer.html#Dataset::setProperty) of the dataset with `setProperty` method:
```